packages/external-db-mssql/lib/mssql_operations.js

```diff
@@ -9,16 +8,13 @@ class DatabaseOperations {
     async validateConnection() {
         try {
             const sql = await this.sql
-            return await sql.query('SELECT 1')
-                            .then(() => { return { valid: true } })
-                            .catch((e) => { return { valid: false, error: notThrowingTranslateErrorCodes(e) } })
-        } catch (err) {
-            switch (err.code) {
-                case 'ELOGIN':
-                    return { valid: false, error: new DbConnectionError(`Access to database denied - probably wrong credentials, sql message: ${err.message}`) }
-                case 'ESOCKET':
-                    return { valid: false, error: new DbConnectionError(`Access to database denied - host is unavailable, sql message: ${err.message}`) }
+            if (sql._connected) {
+                return await sql.query('SELECT 1')
+                    .then(() => { return { valid: true } })
+                    .catch((e) => { return { valid: false, error: notThrowingTranslateErrorCodes(e) } })
             }
+            return sql.connect().then(() => { return { valid: true } }).catch((e) => { return { valid: false, error: notThrowingTranslateErrorCodes(e) } })
```

Comment:
why do we try to connect in this case ?

Reply:
Similar to the query 'select 1' in other implementations, in mssql if we try to query a closed pool the only error we'll get is "connection is closed", and if we'll try to connect we'll get more informative error - like wrong credentials

Reply:
And in the first case (pool is connected), we're trying to query the connected pool and if it's worked, the pool is valid. else there's another error, although the pool is connected.